src/sequence/mod.rs

```diff
@@ -2331,177 +2218,6 @@ mod tests {
 
             // now we converge
             verify_data_convergence(vec![replica1, replica2], dataset_length)?;
-
-        }
-
-
-        #[test]
-        fn proptest_dropped_data_can_be_reapplied_and_we_converge(
```

Comment:
These tests here are not valid since the List (and LSeq) CRDT's expect source ordering. We can't drop or shuffle ops arbitrarily and expect these CRDT's to converge unfortunately.

Reply:
> the List (and LSeq) CRDT's expect source ordering

Has this (somehow) always been the case, or the source ordering requirement is a result of the changes in this PR?

Reply:
I'm concerned about this too, isn't this limitation a big problem for us? I don't see why this limitation now exists as from my understanding this didn't in previous LSeq impl, the only problem was the mutability of Policies, but since we are removing that then I wouldn't have expected ordering to be a problem?

Reply:
It did previously as well, it just wasn't enforced.

Deletes break commutativity,  that is `Insert(3, "A"), Delete(3)` is not the same as `Delete(3), Insert(3, "A")`. You've gotten away with it in the past because LSeq was not doing idempotency checks + you don't yet support deletes.

Idempotency checks are needed again when you have deletes, consider the situation: `Insert(3, "A"), Delete(3)` followed by the originial `Insert(3, "A")` being replayed. You'll find that "A" re-appears  in the list.

With the idempotency checks in place, the replayed insert is a no-op.

> isn't this limitation a big problem for us?

It shouldn't be, we need to check that sequence numbers are strictly incrementing before applying an operation. Otherwise you error back and ask the sender to retry and include the missing operations.

Reply:
I see, we don't allow delete ops in public data, so perhaps we can get away with it and not require the order. For private data we still need to define if we will allow it or not, ideally we don't want different behaviors/logic/checks between private and public though....

Reply:
> With the idempotency checks in place, the replayed insert is a no-op.

Reply:
@dan-da that sounds about right, 

> If so, do we need to add some functionality for (3) into BRB? Something like combine MsgDotNotTheNextDot error with an anti_entropy request?

Not necessary I would say, the eventual anti-entropy rounds should take care of this for us.